packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -139,6 +140,17 @@ enum HeroFlightDirection {
 /// [flightShuttleBuilder] is supplied, its output widget is shown during the
 /// flight transition instead.
 ///
+/// The flightShuttleBulder tree uses [InheritedWidget]s from the route the Hero
+/// animation starts from. If the widget tree in [flightShuttleBuilder] changes
+/// its geometry/appearance based on [InheritedWidget]s that are above the Hero
+/// widget tree, for example if both routes provide [MediaQuery] but with
+/// different paddings, you may see jumps/gaps. If you experience this problem,
```

Comment:
We've solved this problem in this PR, right? Probably use another example, such as:

for example, if there's a custom `InheritedWidget` that provides a theme color and is uses within the hero widget,

Reply:
Yes, I've decided to switch the documentation strategy to what you suggested above.